src/bundle.c

```diff
@@ -403,6 +404,62 @@ static gboolean casync_make_blob(const gchar *idxpath, const gchar *contentpath,
 	return res;
 }
 
+static gboolean generate_incremental_data(RaucManifest *manifest, const gchar *dir, GError **error)
+{
+	GError *ierror = NULL;
+
+	g_return_val_if_fail(manifest, FALSE);
+	g_return_val_if_fail(dir, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	for (GList *elem = manifest->images; elem != NULL; elem = elem->next) {
+		RaucImage *image = elem->data;
+		g_autofree gchar *imagepath = g_build_filename(dir, image->filename, NULL);
+
+		if (!image->incremental)
+			continue;
+
+		if (g_strv_contains((const gchar * const*)image->incremental, "block-hash-index")) {
```

Comment:
Why should we reject unsupported incremental methods? That would break backwards compatibility.

Here we want to check if the methods we support are prepared in the bundle. As soon as we have multiple methods, we want to try them in a specific order. I'll add a comment to the code.

Reply:
I've picked your rework of this function and added a `incremental-test-method` which is ignored for unit tests.